code/datums/ai/basic_mobs/pet_commands/pet_use_targeted_ability.dm

```diff
@@ -22,3 +22,18 @@
 /datum/ai_behavior/pet_use_ability/finish_action(datum/ai_controller/controller, succeeded, ability_key, target_key)
 	. = ..()
 	controller.clear_blackboard_key(target_key)
+
+/datum/ai_behavior/pet_use_ability/then_attack
+
+/datum/ai_behavior/pet_use_ability/then_attack/finish_action(datum/ai_controller/controller, succeeded, ability_key, target_key)
+	. = ..()
+	if(succeeded)
+		controller.queue_behavior(/datum/ai_behavior/basic_melee_attack, target_key, BB_PET_TARGETING_STRATEGY)
```

Comment:
I'm a little iffy about this chaining via typepath (just in terms of the risk of subtype proliferation) but can't think of any particularly good way of doing this with event signals.